src/main/java/ch/vorburger/fswatch/DirectoryWatcherImpl.java

```diff
@@ -52,12 +55,27 @@
 
     protected final WatchService watcher = FileSystems.getDefault().newWatchService(); // better final, as it will be accessed by both threads (normally OK either way, but still)
     protected final Thread thread;
-
-    // protected because typical code should use the DirectoryWatcherBuilder instead of this directly
+    protected final List<ChangeKind> changeKindsList = new ArrayList<>();
+    
+    /*
+     * Retained the existing constructor and it will retain the old behaviour of responding only to MODIFIED and DELETED ChangeKinds
+     */
```

Comment:
just remove this?  I see you're explaining what you are doing here to me, but you don't have to, the tests should ensure that the old behaviour is retained.  Imagine yourself as another future user coming across this later - the history of how we got here won't matter to her.